src/core/mac/sub_mac.cpp

```diff
@@ -500,6 +500,19 @@ void SubMac::HandleTransmitDone(TxFrame &aFrame, RxFrame *aAckFrame, Error aErro
     bool ccaSuccess = true;
     bool shouldRetx;
 
+#if OPENTHREAD_RADIO
+    if (aFrame.GetChannel() != aFrame.GetRxChannelAfterTxDone() && mRxOnWhenIdle)
```

Comment:
BTW, what is reason to limit this behavior to when we have `mRxOnWhenIdle`?

Reply:
The intention is to prevent the coprocessor being put into receive state while it should sleep after tx done(when rx-on-when-idle is false).

Reply:
Sounds good. Thanks.